packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1056,8 +1057,26 @@ export class ContainerRuntime
 			(schema) => {
 				runtime.onSchemaChange(schema);
 			},
+			minVersionForCollab,
 		);
 
+		// We check the document's metadata to see if there is a minVersionForCollab. If it's not an existing document or
+		// if the document is older, then it won't have one. If it does have a minVersionForCollab, we check if it's greater
+		// than this client's runtime version. If so, we log a telemetry event to warn the customer that the client is outdated.
+		// Note: We only send a warning because we already ensured that this client **can** understand the existing document's
+		// schema (part of the `DocumentsSchemaController` constructor flow above). However, we still want to issue a warning to
```

Comment:
```suggestion
		// Note: We only send a warning because we already found that this client **can** understand the existing document's
		// schema (the `DocumentsSchemaController` constructor throws otherwise). However, we still want to issue a warning to
```

Reply:
Future: it's a little surprising to me that a constructor throwing is a key piece of control flow.  Maybe would be more clear to move to private constructor and a function `createDocumentSchemaControllerIfCompatible` or something?

In the meantime, maybe quickly update the constructor's doc comment to point out it will throw in XYZ case(s).